### PR TITLE
Update zstd magic number to comply with the latest zip specification

### PR DESCRIPTION
**Pull request by fcharlie**
See: https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT

```txt
        0 - The file is stored (no compression)
        1 - The file is Shrunk
        2 - The file is Reduced with compression factor 1
        3 - The file is Reduced with compression factor 2
        4 - The file is Reduced with compression factor 3
        5 - The file is Reduced with compression factor 4
        6 - The file is Imploded
        7 - Reserved for Tokenizing compression algorithm
        8 - The file is Deflated
        9 - Enhanced Deflating using Deflate64(tm)
       10 - PKWARE Data Compression Library Imploding (old IBM TERSE)
       11 - Reserved by PKWARE
       12 - File is compressed using BZIP2 algorithm
       13 - Reserved by PKWARE
       14 - LZMA
       15 - Reserved by PKWARE
       16 - IBM z/OS CMPSC Compression
       17 - Reserved by PKWARE
       18 - File is compressed using IBM TERSE (new)
       19 - IBM LZ77 z Architecture 
       20 - deprecated (use method 93 for zstd)
       93 - Zstandard (zstd) Compression 
       94 - MP3 Compression 
       95 - XZ Compression 
       96 - JPEG variant
       97 - WavPack compressed data
       98 - PPMd version I, Rev 1
       99 - AE-x encryption marker (see APPENDIX E)

```

### CODE REVIEW

File: mz_strm_zstd.c [hunk 1]
```diff
@@ -160,7 +160,7 @@ int32_t mz_stream_zstd_read(void *stream, void *buf, int32_t size) {
         zstd->total_in += in_bytes;
         zstd->total_out += out_bytes;
 
-    } while (zstd->in.pos < zstd->in.size && zstd->out.pos < zstd->out.size);
```
Comment (nmoinvaz) [inline mz_strm_zstd.c]: Why this change?

**Comment by nmoinvaz** [comment]
We can just get rid of 0x20 value.

File: mz_strm_zstd.c (hunk 1 above)
Comment (fcharlie) [reply mz_strm_zstd.c]: This is a previous modification and is currently consistent with upstream.

**Comment by nmoinvaz** [comment]
Thanks!

**Comment by gvollant** [comment]
I think this is a very bad idea to not support 20 magic when decompress

I'm of course OK to create new zipfile with magic 93.
But compatibility is always better than error.
So I prefer support both 20 and 93 for decompress zstd

**Comment by gvollant** [comment]
@nmoinvaz @fcharlie

**Comment by fcharlie** [comment]
@gvollant The key to this question is how many compression software/libraries support decompressing zip (zstd-20), but at present, I have not found any software or libraries that have been released to support magic number 20, so regarding compatibility, We don't need to consider the compatibility of magic number 20. Of course, if there is software that widely uses 20 as a zip(zstd) magic number, you can tell me. Thanks

**Comment by fcharlie** [comment]
@gvollant Another problem is that the specifications in the zip format are in the hands of PKWARE, and interested developers cannot participate in the formulation of the specifications. The changes may be arbitrary.

**Comment by nmoinvaz** [comment]
It looks like they changed the spec very quickly so it is doubtful that anybody used 20. Maybe a mistake on PKWARE’s part.
